pymilvus/orm/connections.py

```diff
@@ -293,13 +344,19 @@ def connect(
             >>> connections.connect("test", host="localhost", port="19530")
         """
 
+        # kwargs_copy is used for auto reconnect
+        kwargs_copy = copy.deepcopy(kwargs)
+
         def connect_milvus(**kwargs):
             gh = GrpcHandler(**kwargs)
 
             t = kwargs.get("timeout")
             timeout = t if isinstance(t, (int, float)) else Config.MILVUS_CONN_TIMEOUT
 
             gh._wait_for_channel_ready(timeout=timeout)
+            gh.register_state_change_callback(
+                ReconnectHandler(self, alias, kwargs_copy).reconnect_on_idle
+            )
```

Comment:
please add a flag to control whether to add the callback or not, for instance, reset_connection=True.

Do not set it as default bahaviour to aviod unknown bugs. After all It's a temporay solution.

Reply:
updated